UIMod/onboard_bundled/scripts/autostart.sh

```diff
@@ -1,3 +1,61 @@
 #!/bin/bash
+set -e
 
-# not yet implemented
\ No newline at end of file
+# Check if running as root to prevent installing a service as root
+if [[ $(id -u) = 0 ]]; then
+  echo "For security reasons, it is not recommended to run this service as a root user."
+  exit 1
+fi
+
+# Check if systemd is available
+if ! command -v systemctl &> /dev/null; then
+  echo "Error: systemd is not available on this system."
+  exit 1
+fi
+
+# Determine the full path of this script
+SCRIPT_PATH=$(readlink -f "$0")
+
+# Determine the base directory and locate the StationeersServerControl binary
+BASEDIR=$(dirname "$SCRIPT_PATH")
+if [[ -z "$BASEDIR" || ! -d "$BASEDIR" ]]; then
+  echo "Error: Could not determine base directory."
+  exit 1
+fi
+
+# Find the last modified SSUI binary if multiple exist
+SSUI_BINARY=$(ls -t "$BASEDIR"/StationeersServerControlv* 2>/dev/null | head -n 1)
```

Comment:
Once upon a time, I learned the hard way that ls is not script safe - **would** like find more.

```suggestion
SSUI_BINARY=$(find "$BASEDIR" -maxdepth 1 -name 'StationeersServerControlv*' -type f -executable -print0 | xargs -0 ls -t | head -n 1)
```

could also add something like this below

```
BINARY_COUNT=$(find "$BASEDIR" -maxdepth 1 -name 'StationeersServerControlv*' -type f -executable | wc -l)
if [[ "$BINARY_COUNT" -gt 1 ]]; then
  echo "Warning: Multiple StationeersServerControl binaries found. Using the most recent: $SSUI_BINARY"
fi`
```

Reply:
Good with that, honestly I think I was sleepy and grumpy at copilot for other silly suggestions, this is a good one now that I look at it after my morning coffee